### PR TITLE
docs: update godoc uses of "provider" to "driver" or "service" in top-level .go files

### PR DESCRIPTION
**Pull request by vangent**
Updates #2548.

This updates most of the top-level `.go` files in each portable type.

### CODE REVIEW

File: docstore/doc.go [hunk 1]
```diff
@@ -134,7 +134,7 @@
 // Times deserve special mention. Docstore can store and retrieve values of type
 // time.Time, with two caveats. First, the timezone will not be preserved. Second,
 // Docstore guarantees only that time.Time values are represented to millisecond
-// precision. Many providers will do better, but if you need to be sure that times
+// precision. Many drivers will do better, but if you need to be sure that times
```
Comment (polinasok) [inline docstore/doc.go]: services?

File: docstore/doc.go [hunk 2]
```diff
@@ -231,8 +231,8 @@
 //   iter := coll.Query().Where("Author.Last", "=", "Bulgakov").Limit(3).Get(ctx)
 //
 // You can make multiple Where calls. In some cases, parts of a Where clause may be
-// processed on the client rather than natively by the provider, which may have
-// performance implications for large result sets. See the provider-specific package
+// processed in the driver rather than natively by the backend service, which may have
```
Comment (polinasok) [inline docstore/doc.go]: I believe "backing service" is more common with cloud/apps.

File: docstore/doc.go (hunk 1 above)
Comment (vangent) [reply docstore/doc.go]: Sure, done.

File: docstore/doc.go (hunk 2 above)
Comment (vangent) [reply docstore/doc.go]: Done.

**Pull request merged**